### DISCUSSION TITLE
Change panels order

### DISCUSSION THREAD

**Discussion by kds112** [devemux86/cruiser#318] [Q&A] [answered]
The update 3.0.32 was a huge step in functionality, thanks al lot! I have two minor suggestions for further improvement. Is it possible to sort the upright panels (clock, battery, satellites, navigation instructions, etc) in a different order? Maybe the settings for the panels should be moved to general map settings (they are now in the navigation section)? There you could sort them and you could select to show them always/never/during navigation/etc.

For the second suggestion I open another discussion…

**Comment by devemux86** [comment]
@kds112 thanks for the idea.

I already have in my list to provide a way to select the order of all these info panels.

**Comment by devemux86** [comment]
To avoid complicating things too much we should probably follow some rules:
- Direction panels at top left
- ~Distance / time at top right, fixed?~
Or should they participate in the panels arrangement?
- Nearby favorites + POI at bottom of all other panels
(for easy discovery as their icons and number change during navigation)

Then we can change the arrangement of all other panels.
Any thoughts?

**Reply by WolfgangZ42** [reply]
> Any thoughts?

Yes ... not really important! I'm operating Cruiser in two modes: Navigate and Follow. To have the same "layout" in both cases, it could be fine to put **_current speed_** and **_gps recording_** (if enabled) at top right. 
Pictures:
<img width="300" src="https://github.com/devemux86/cruiser/assets/121806238/a9762ef7-8909-4653-af78-f4da00947adb"/> 
<img width="300" src="https://github.com/devemux86/cruiser/assets/121806238/8153a584-526d-4e08-8c21-00f3e048d51e"/>

**Reply by devemux86** [reply]
So we probably need arrangement also for the distance + time panels.

Then we can put the speed + recording panels at the top of them
and your images will become similar (for these 2 panels).

**Reply by kds112** [reply]
Sounds good! I prefer to have the panels top right, but I am only using clock, battery altitude and satellites...

**Reply by devemux86** [reply]
Possible different arrangement of the navigation UI can be examined in a separate discussion.

We also have a discussion for a possible different UI in landscape orientation:
- https://github.com/devemux86/cruiser/discussions/242

Here it is about the arrangement of the panels at top right with the existing UI.

**Comment by devemux86** [comment]
Video with the new settings:
<table>
<tr>
<td width="50%"><video src="https://github.com/devemux86/cruiser/assets/3484020/7b1e7b48-b79e-4336-b59b-c09049f3180b"/></td>
<td width="50%"><img src="https://github.com/devemux86/cruiser/assets/3484020/54d6182a-5f0b-47ad-85fa-87cf70adc2dc"/></td>
</tr>
</table>

**Reply by Ciutnik** [reply]
Although this is not a necessary function for me, I admit that the order change implemented in this way makes a great impression 👍

This reserves a long press on the panel for other functions in the future!

**Reply by kds112** [reply]
Looks gorgeous! 👍👍👍

**Reply by devemux86** [reply]
I thought if we could move the on / off switches in the new screen,
but for now keep it simple and have the on / off switches in settings.

But indeed this new screen could host more functions in the future.

Also, the on / off switches will now be multi-selected so users can handle them more freely:
- [x] Navigation
- [x] Follow location
- [x] Browse map

**Reply by Ciutnik** [reply]
> I thought if we could move the on / off switches in the new screen

That was the first thing that occurred to me, that there was no on/off switch here. But I didn't mention it because it's too obvious for you not to have thought of it too 😀

**Comment by Leeb64** [comment]
Changing the panel order is good!
What's not so good is that I have to make the changes separately for each route profile...quite complicated...

**Reply by devemux86** [reply]
All app settings can be made as global for all profiles.
We just need to decide which settings we want as global.

Currently the global settings are:
- Language
- Extended settings
- Storage folder

**Reply by Leeb64** [reply]
Panel order would make sense!
You don't want to "rethink" at every other profile?

**Reply by WolfgangZ42** [reply]
> quite complicated

I guess you will not change it constantly - so it's just an one-time-effort (and not really an effort). Why should we loose flexibility?

If the panel order is integrated into the enabling/disabling of panels it should stay in profile setting anyway.

**Reply by kds112** [reply]
Maybe you could mark the extended section as "(global)"? Then it is clearly that all other settings are profile-dependend...

**Reply by devemux86** [reply]
@kds112 what do you mean?

**Reply by kds112** [reply]
I just propose to label these settings as "global":
Language
Extended settings
Storage folder

**Reply by devemux86** [reply]
In the past we also labeled the settings that restart the app in order to be activated (e.g. language, theme).
But I'm not so sure if most users need that much information and won't be confused by it.

**Comment by devemux86** [comment, marked as answer]
Change panels order is available in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
